tests/data/scad/misc/lookup-tests.scad

```diff
@@ -15,3 +15,23 @@ indices = [-20,-10,-9.9, -0.5, 0, 0.3, 1.1, 10, 10.1];
 for (i=[0:len(indices)-1]) {
   echo(lookup(indices[i], table));
 }
+
+table2DVector =
+        [[-1,  [-5, 5]],
+         [-10, [-55, 55]],
+         [0,    [0, 0]],
+         [1,    [3, -3]],
+         [10,   [333, -333]]];
+for (i=[0:len(indices)-1]) {
+  echo(lookup(indices[i], table2DVector));
+}
+
+table3DVector =
+        [[-1,  [-5, 5, -1]],
+         [-10, [-55, 55, -10]],
+         [0,    [0, 0, 0]],
+         [1,    [3, -3, 1]],
+         [10,   [333, -333, 10]]];
+for (i=[0:len(indices)-1]) {
+  echo(lookup(indices[i], table3DVector));
+}
```

Comment:
Add tests for error cases:

- [ ] key is not a number
- [ ] table is not a vector
- [ ] entry in table is not a vector
- [ ] entry in table has size other than 2
- [ ] entry[0] is not a number
- For the two entries bracketing a selected entry:
- [ ] entry[1] is neither a number nor a vector
- [ ] entry[1] is a vector containing values that are neither numbers nor vectors.
- [ ] the two entry[1] values are not consistent types - one is a number and the other is a vector.
- [ ] the two entry[1] values are both vectors, but they have different lengths.
- [ ] the two entry[1] values are both vectors, but one has an entry that is a number while the corresponding entry in the other is a vector.
- For the high or low entry, selected when the entry is out of bounds:
- [ ] [1] is not a number or vector.
- [ ] [1] is a vector, but it contains something other than a number or vector.  (Note:  recursively checking this is probably out of scope.)

Note:  a not-particularly-silly way to handle the type mismatch cases on the boundary cases might be to multiply them by 1.0.  That will do nothing if they are OK, and otherwise will trigger the same error handling that the interpolation cases would trigger, so you would at least sort of consistently get undef or warnings.

I suspect that most of those cases will yield undef and that a few will yield warnings.  I'm not looking for any behavior in particular, just that they not yield apparently-valid values and that they not yield crashes.